modules/siddhi-core/src/main/java/io/siddhi/core/util/transport/TemplateBuilder.java

```diff
@@ -104,17 +109,43 @@ private MessageFormat parseTextMessage(StreamDefinition streamDefinition, String
             if (m.group(1) != null) {
                 int attrIndex = attributes.indexOf(m.group(1).replaceAll("\\p{Ps}", "").replaceAll("\\p{Pe}", ""));
                 if (attrIndex >= 0) {
-                    m.appendReplacement(result, String.format("{%s}", attrIndex));
+                    m.appendReplacement(result, String.format("{{%s}}", attrIndex));
                 } else {
                     throw new NoSuchAttributeException(String.format("Attribute : %s does not exist in %s.",
                             m.group(1), streamDefinition));
                 }
             } else {
-                m.appendReplacement(result, " '" + m.group() + "' ");
+                m.appendReplacement(result, m.group() + "");
             }
         }
         m.appendTail(result);
-        return new MessageFormat(result.toString());
+        return result.toString();
+    }
+
+    private String formatMessage(Object[] outputData) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < splitTemplateArray.length; i++) {
+            if (i % 2 == 0) {
+                stringBuilder.append(splitTemplateArray[i]);
+            } else {
+                stringBuilder.append(outputData[Integer.parseInt(splitTemplateArray[i])]);
+            }
+        }
+        return stringBuilder.toString();
+    }
+
+    private void validateTemplateArrayAttributePositions(String[] splitTemplateArray,
+                                                         StreamDefinition streamDefinition) {
+        for (int i = 0; i < splitTemplateArray.length; i++) {
+            if (i % 2 != 0) {
+                try {
+                    Integer.parseInt(splitTemplateArray[i]);
```

Comment:
Shall we add the ints to an array and use that or make splitTemplateArray an Object[]